amp/WEB-INF/src/org/digijava/module/aim/util/ActivityVersionUtil.java

```diff
@@ -668,11 +681,15 @@ private static void processVersionableSimple(AmpActivityVersion ampActivityOne,
 
     public static Map<String, Map<String, List<CompareOutput>>> getOutputCollectionGrouped() {
 
+        StopWatch stopWatch = new StopWatch();
+        System.out.println("STARTING STOPWATCH");
+        stopWatch.start();
+        
         Map<String, Map<String, List<CompareOutput>>> listOfOutputCollectionGrouped = new HashMap<>();
-
         //Use lambda through the accept method from java consumer functional interface
         // to create listOfOutputCollectionGrouped
-        AuditLoggerUtil.getListOfActivitiesFromAuditLogger().forEach((Object[] activityObj) -> {
+            
+        AuditLoggerUtil.getListOfActivitiesFromAuditLogger().parallelStream().forEach((Object[] activityObj) -> {
```

Comment:
unless im seeing something wrong the code is only processing the first record. We need to use CompletableFutures and stream joining. I will create a sample, and Ill schedule a call for next wednesday